tests/integration/passthrough-test.js

```diff
@@ -202,3 +202,45 @@ test('it can passthrough other-origin hosts', function(assert) {
   });
 });
 
+
+test('passthrough requests follow normal precedence rules 1', function(assert) {
+  assert.expect(1);
+  var done1 = assert.async();
+  var server = this.server;
+
+  server.loadConfig(function() {
+    this.urlPrefix = '/vet/123';
+    this.passthrough('/pets');
+    this.get('/pets', function() { return 123; });
+  });
+
+  $.ajax({
+    method: "GET",
+    url: "/vet/123/pets",
+    error: function() {
+      assert.ok(true);
+      done1();
+    }
+  });
+});
+
+test('passthrough requests follow normal precedence rules 2', function(assert) {
+  assert.expect(1);
+  var done1 = assert.async();
+  var server = this.server;
+
+  server.loadConfig(function() {
+    this.urlPrefix = '/vet/123';
+    this.get('/pets', function() { return 123; });
+    this.passthrough('/pets');
```

Comment:
In this case, I'd expect the first handler (the mirage handler `.get('/pets')`) to "win", since it's defined first. I think you may need to update your strategy here to accommodate this.

I don't want you to have to re-invent Pretender's route handler registry... perhaps this is something we should fix upstream in Pretender itself?